File: Assets/My Assets/Scripts/Gameplay/InstantActionLoad.js
var neutralStation : GameObject;
var player : GameObject;
var enemyShip : GameObject;
var spawnDistance : float = 400;
private var asteroidGen : GameObject;
private var eCommander : GameObject;
private var stationCount;
function setReferences(){ //Update our references (this saves frames)
	asteroidGen = GameObject.FindGameObjectWithTag("AsteroidGen");
	eCommander = GameObject.FindGameObjectWithTag("EAICOM");
}
function setSettings(){ //Set Instant Action Preferences from the menu
	asteroidGen.GetComponent("generateField").asteroids = PlayerPrefs.GetInt("IA_ASTEROIDS");
	asteroidGen.GetComponent("generateField").spawnRange = PlayerPrefs.GetInt("IA_DENSITY");
	eCommander.GetComponent("AICommander").myType = PlayerPrefs.GetInt("IA_AITYPE");
	stationCount = PlayerPrefs.GetInt("IA_STATIONS");
}
function Awake(){//On scene load, do this:
	setReferences();
	setSettings();
	spawnMap();
}
function spawnMap(){
	for(i = 0;i< stationCount-1;i++){
		if(i ==0){
			player.transform.position = Vector3(-20,0,0);
		}
		Instantiate(neutralStation,Vector3(i*spawnDistance,0,0),Quaternion.identity);
		if(i == stationCount-1){
			enemyShip.transform.position = Vector3(i*spawnDistance+10,0,0);
		}
	}
}

File: Assets/My Assets/Scripts/Gameplay/InstantActionLoad.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class InstantActionLoad : MonoBehaviour
{
    public GameObject neutralStation;
    public GameObject player;
    public GameObject enemyShip;
    public float spawnDistance;
    public bool localMP;
    private GameObject asteroidGen;
    private GameObject eCommander;
    private object stationCount;
    public virtual void setReferences() //Update our references (this saves frames)
    {
        this.asteroidGen = GameObject.FindGameObjectWithTag("AsteroidGen");
        this.eCommander = GameObject.FindGameObjectWithTag("EAICOM");
    }

    public virtual void setSettings() //Set Instant Action Preferences from the menu
    {
        this.asteroidGen.GetComponent<generateFields>().asteroids = PlayerPrefs.GetInt("IA_ASTEROIDS");
        this.asteroidGen.GetComponent<generateFields>().spawnRange = PlayerPrefs.GetInt("IA_DENSITY");
        if (this.eCommander)
        {
            this.eCommander.GetComponent<AICommander>().myType = (CommanderType)PlayerPrefs.GetInt("IA_AITYPE");
        }
        this.stationCount = PlayerPrefs.GetInt("IA_STATIONS");
    }

    public virtual void Awake()//On scene load, do this:
    {
        this.setReferences();
        this.setSettings();
        if (this.localMP == false)
        {
            this.spawnMap();
        }
    }

    public virtual void spawnMap()
    {
        int i = 0;
        while (i < (((int) this.stationCount) - 1))
        {
            if (i == 0)
            {
                this.player.transform.position = new Vector3(-20, 0, 0);
            }
            UnityEngine.Object.Instantiate(this.neutralStation, new Vector3(i * this.spawnDistance, 0, 0), Quaternion.identity);
            if (i >= (((int) this.stationCount) - 2))
            {
                this.enemyShip.transform.position = new Vector3((i * this.spawnDistance) + 10, 0, 0);
            }
            i++;
        }
    }

    public InstantActionLoad()
    {
        this.spawnDistance = 400;
    }

}